File: 프로그래머스/unrated/181921. 배열 만들기 2/배열 만들기 2.java
import java.util.ArrayList;

class Solution {
    public int[] solution(int l, int r) {
        ArrayList<Integer> result = new ArrayList<>();

        for (int num = l; num <= r; num++) {
            if (isOnlyZeroAndFive(num)) {
                result.add(num);
            }
        }

        if (result.isEmpty()) {
            return new int[]{-1};
        }

        return result.stream().mapToInt(Integer::intValue).toArray();
    }

    private boolean isOnlyZeroAndFive(int num) {
        while (num > 0) {
            int digit = num % 10;
            if (digit != 0 && digit != 5) {
                return false;
            }
            num /= 10;
        }
        return true;
    }
}

File: 프로그래머스/0/181921. 배열 만들기 2/배열 만들기 2.py
def solution(l, r):
    result = []
    
    for num in range(l, r+1):
        if is_only_zero_and_five(num):
            result.append(num)
    
    if not result:
        return [-1]
    
    return result

def is_only_zero_and_five(num):
    while num > 0:
        digit = num % 10
        if digit != 0 and digit != 5:
            return False
        num //= 10
    return True
